src/broadcast/broadcast.rs

```diff
@@ -157,9 +157,7 @@ impl<N: NodeIdT> Broadcast<N> {
         let mut shards: Vec<&mut [u8]> = shards_iter.collect();
 
         // Construct the parity chunks/shards
-        self.coding
-            .encode(&mut shards)
-            .expect("incorrect shard size or number");
+        self.coding.encode(&mut shards)?;
```

Comment:
Is this an error that can realistically occur, that we want to give the user ability to handle? Or is this just to satisfy the audit?

Reply:
I _think_ it shouldn't be possible, looking at [the docs](https://docs.rs/reed-solomon-erasure/3.1.1/reed_solomon_erasure/enum.Error.html), but they're not 100% clear about this.
If we keep this as `expect`, we should probably remove all Reed-Solomon-related error variants.

Reply:
Well, since the user expects no Reed-Solomon-Errors for any arbitrary input they put in, we should not expose errors there to the user, right?

We should probably use `expect`s and `assert`s there that indicate any time they are a triggered, it is 100% certain to be a bug in either of the two libraries.